code/Model/SiteTree.php

```diff
@@ -2713,6 +2713,26 @@ public function getPageLevel() {
 	}
 
 	/**
+	 * @return string
+	 */
+	public function getControllerName() {
+		if ($this->class === SiteTree::class) {
+			$controller = ContentController::class;
+		} else {
+			$ancestry = ClassInfo::ancestry($this->class);
+			while ($class = array_pop($ancestry)) {
+				if (class_exists($class . "_Controller")) {
+					break;
+				}
+			}
+
+			$controller = ($class !== null) ? "{$class}_Controller" : ContentController::class;
```

Comment:
Why strict type check $class? Surely empty `$class` shouldn't be accepted as valid?